### PR TITLE
feat: support model parameters

### PR DESCRIPTION
**Pull request by arjunattam**

### CODE REVIEW

**Comment by arjunattam** [comment]
updated based on our discussion, @saikatmitra91. i will push schema to cloudflare after #74

File: packages/ai/src/utils/index.ts [hunk 1]
```diff
@@ -5,3 +5,34 @@ export function replacePlaceholders(string: string, obj: any) {
     return obj[key];
   });
 }
+
+function isKnownParameter(paramName: string) {
```
Comment (saikatmitra91) [inline packages/ai/src/utils/index.ts:9]:
```suggestion
function isReservedParameter(paramName: string) {
```

File: packages/ai/src/utils/index.ts [hunk 2]
```diff
@@ -5,3 +5,34 @@ export function replacePlaceholders(string: string, obj: any) {
...
+  const knownParameters = [
```
Comment (saikatmitra91) [inline packages/ai/src/utils/index.ts:10]:
```suggestion
  const reservedParameters = [
```

File: packages/ai/src/utils/index.ts [hunk 3]
```diff
@@ -5,3 +5,34 @@ export function replacePlaceholders(string: string, obj: any) {
...
+    "frequency_penalty",
+    "logprobs",
+    "n",
+    "presence_penalty",
+    "response_format",
+    "seed",
+    "stop",
+    "top_logprobs",
+  ];
+  const otherFieldsToSkip = ["model", "messages"];
```
Comment (saikatmitra91) [inline packages/ai/src/utils/index.ts:23]: if we  rename the list to reserved, dont have to keep this as separate array ?

File: packages/ai/src/utils/index.ts [hunk 4]
```diff
@@ -5,3 +5,34 @@ export function replacePlaceholders(string: string, obj: any) {
...
+}
+
+export function getPassthroughParams(body: { [key: string]: any }) {
+  let passthroughParams: { [key: string]: any } = {};
+
+  Object.keys(body)
+    .filter((key) => !isKnownParameter(key))
+    .forEach((key) => {
+      passthroughParams[key] = body[key];
+      body[key] = undefined;
```
Comment (saikatmitra91) [inline packages/ai/src/utils/index.ts:34]: why do we need to do this ?

**Review by saikatmitra91 — CHANGES_REQUESTED** [review]

**Review by saikatmitra91 — APPROVED** [review]

**Pull request merged**